### ISSUE TITLE
Rscript reports not enough space, though there should be and RStudio has no problem

### ISSUE THREAD

**Issue by hack-r** [BioStatMatt/sas7bdat.parso#7]
Could I ask you to check out this issue on StackOverflow?

http://stackoverflow.com/questions/38837517/sas7bdat-can-import-files-in-rstudio-but-says-not-enough-space-with-rscript

I have an R script that imports several files from SAS using the sas7bdat.parso.

There's no problem running this in RStudio. However, when I try to run this with Rscript.exe as a scheduled task, the task does not complete.

The error message in the log is:

```
Error in s7b2csv(file, tempfile()) : There is not enough space on the disk

Calls: fread.sas7bdat.parso -> read.sas7bdat.parso -> READ_FUNC -> s7b2csv Execution halted
```

How can I resolve this?

I do have a lot of drives in the system but from what I can tell the drives involved have several hundred GB free as do the drives which are not involved.

**Comment by BioStatMatt** [comment]
Can you try to modify the R code such that 'tempfile()' is replaced by a
string that gives an alternate file location? I'm trying to determine
whether it's a tempfile() issue or something else.

-Matt

**Comment by hack-r** [comment]
Thanks Matt. Yes I will give it a try. I am away from that computer for the day but will do it first thing in the morning.

Sent from my iPhone

**Comment by hack-r** [comment]
Sorry for the delayed response.

That resulted in the same outcomes as before -- it worked in RStudio but not when run with Rscript.

In the code I made my edit to the `fread.sas7bdat.parso` lines, e.g.

`result <- fread.sas7bdat.parso("G:\\temp\\tmp.sas7bdat", tempfile("G://"),sep="auto")
`

but it the log it calls out the internal `s7b2csv` function call:

> Error in s7b2csv(file, tempfile()) : 
>   There is not enough space on the disk
> Calls: fread.sas7bdat.parso -> read.sas7bdat.parso -> READ_FUNC -> s7b2csv

So perhaps it's not passing that `tempfile` option to `s7b2csv`?

I did however confirm that the `TMP` and `TMPDIR` directories for my R session have over 250 GB free and should need no more than 1 GB at most.
